presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -369,7 +381,12 @@ public CounterStat getHighMemorySplitSource()
                     // verify partition is not marked as non-readable
                     String partitionNotReadable = partition.getParameters().get(OBJECT_NOT_READABLE);
                     if (!isNullOrEmpty(partitionNotReadable)) {
-                        throw new HiveNotReadableException(tableName, Optional.of(partName), partitionNotReadable);
+                        if (!shouldIgnoreUnreadablePartition(session)) {
+                            throw new HiveNotReadableException(tableName, Optional.of(partName), partitionNotReadable);
+                        }
+                        warningCollector.add(new PrestoWarning(HiveWarningCode.PARTITION_NOT_READABLE,
```

Comment:
- move the first param to it's own line
- static import `PARTITION_NOT_READABLE`
- Can we add a unit test for this path?